scipy/stats/_continuous_distns.py

```diff
@@ -6123,6 +6125,43 @@ def _stats(self, b, moments='mv'):
     def _entropy(self, c):
         return 1 + 1.0/c - np.log(c)
 
+    def fit(self, data, *args, **kwds):
+        data = np.asarray(data)
+
+        floc = kwds.pop('floc', None)
+
+        # borrowed from distn_infrastructure
+        # puts shape into f0
+        if self.shapes:
+            shapes = self.shapes.replace(',', ' ').split()
+            for j, s in enumerate(shapes):
+                key = 'f' + str(j)
+                names = [key, 'f' + s, 'fix_' + s]
+                val = _get_fixed_fit_value(kwds, names)
+                if val is not None:
+                    kwds[key] = val
+
+        shape = kwds.pop('f0', None)
+        fscale = kwds.pop('fscale', None)
+
+        _check_fit_input_parameters(data, args,
+                                    kwds, fixed_param=(shape, floc, fscale))
+
+        if floc is None or fscale is not None or np.any(data - floc < 0):
+            if floc is not None:
```

Comment:
You just removed these from `kwds`, now you're putting them back. Why not just use `kwds.get` instead of `kwds.pop` above?

Reply:
The reason for that nastiness was that I did not want to leave `floc` `f0` or `fscale` in kwds for when the check is done for unknown items in the dictionary. However, all kwds should still be passed to the super method.

For now to eliminate all of this I just created a copy of kwds that is never modified that is passed to the super method, but we can discuss other strategies.